File: Programmers/87946.py
def solution(k, dungeons):
    dungeons.sort()
    answer = []
    n = len(dungeons)
    visited = [0] * 8

    def backtracking(kk, cnt):
        for i in range(n):
            if visited[i] == 0:
                if kk >= dungeons[i][0]:
                    visited[i] = 1
                    backtracking(kk - dungeons[i][1], cnt + 1)
                    visited[i] = 0
                else:
                    answer.append(cnt)
                    return

        answer.append(cnt)

    backtracking(k, 0)
    return max(answer)

File: Programmers/87946.js
// 완전탐색 - 피로도
function solution(k, dungeons) {
    let maxCnt = 0;
    let visited = new Array(10).fill(false);

    function dfs(cnt, left) {
        for (let i = 0; i < dungeons.length; i++) {
            if (visited[i] === false && left >= dungeons[i][0]) {
                visited[i] = true;
                dfs(cnt + 1, left - dungeons[i][1])
                visited[i] = false;
            }
        }
        maxCnt = Math.max(maxCnt, cnt);
    }
    dfs(0, k);
    return maxCnt;
}